### ISSUE TITLE
Soundcore Q20i

### ISSUE THREAD

**Issue by DasJott** [JordanViknar/Noiseclapper#21]
Hey!
Thank you for this extension - even though it doesn't work for my Q20i.
I had a look into your code and found out that you're listing devices in common.js. I just added my model as "soundcore Q20i" to `supportedDeviceNames` and now it is found as a devide.
However, when I send signals to it, it simply does not respond.

How did you find out all these hex-signals you're listing in common.js?
Maybe only those signatures  are different for this model? How can I find that out?

Would like to help with adding more soundcore models, if I can.

Already played around with `bluetoothctl` a bit, but that one is poorly documented on first sight.

**Comment by JordanViknar** [comment]
Before I answer your question, I figure I should say this : 

> found out that you're listing devices in `common.js`

The extension is not meant to be modified directly like this, since it's transpiled from TypeScript.

To "hint" that the actual TypeScript source code should be modified, not the output JavaScript files, I had originally configured `tsconfig.json` to strip the source code comments during JavaScript conversion. That was also done for the sake of general efficiency.

Considering this isn't very effective, and only really makes it harder for people to modify very quickly the extension, I figure I should rather turn that feature off.

> How did you find out all these hex-signals you're listing in common.js ?
> Maybe only those signatures are different for this model ? How can I find that out ?

**TL;DR : I didn't.**

For a good while, Noiseclapper wasn't on the GNOME Extensions website because it had an older version of [this library](https://github.com/KillerBOSS2019/SoundcoreDesktop) bundled inside, and that's what it used to send signals.
Eventually, the library was removed and the extension directly handles signals itself now, relying on a small Python command to forward them to the headphones.

However... I never did learn how the signals used were obtained in the first place, so I cannot help you with this myself. I'm sorry.

Perhaps, you could try checking out [OpenSCQ30](https://github.com/Oppzippy/OpenSCQ30), it *might* have working support for your device.
If not, that's always another project that could also use some help getting support for your headphones.

**Comment by DasJott** [comment]
Thank you nonetheless for your reply!

**Comment by JordanViknar** [comment]
Realizing now that I completely forgot to apologize for my answer's *biiiiig* delay.
Sorry, that was because I had a bit more work from school on my hands than usual for the past 2 weeks.
